encodings/alp/src/alp.rs

```diff
@@ -77,27 +77,40 @@ pub trait ALPFloat: Float + 'static {
             .iter()
             .enumerate()
             .map(|(i, v)| {
-                let encoded =
-                    (*v * Self::F10[exp.e as usize] * Self::IF10[exp.f as usize]).fast_round();
-                let decoded = encoded * Self::F10[exp.f as usize] * Self::IF10[exp.e as usize];
-
-                if decoded == *v {
-                    if let Some(e) = encoded.as_int() {
-                        prev = e;
-                        return e;
+                match Self::encode_single(*v, &exp) {
+                    Ok(fi) => {
+                        prev = fi;
+                        fi
+                    }
+                    Err(exc) => {
+                        exc_pos.push(i as u64);
+                        exc_value.push(exc);
+                        // Emit the last known good value. This helps with run-end encoding.
+                        prev
                     }
                 }
-
-                exc_pos.push(i as u64);
-                exc_value.push(*v);
-                // Emit the last known good value. This helps with run-end encoding.
-                prev
             })
             .collect_vec();
 
         (exp, encoded, exc_pos, exc_value)
     }
 
+    #[inline]
+    fn encode_single(value: Self, exponents: &Exponents) -> Result<Self::ALPInt, Self> {
```

Comment:
I thought I needed this function but in the end I don't. However, I kept it here since it's a useful logic block